lib/libc/string/memchr.c

```diff
@@ -39,8 +41,17 @@ memchr(const void *src, int c, size_t n)
 {
 	const unsigned char *s = src;
 	c = (unsigned char)c;
-// XXX-BD: this path produces out of bounds cld's on CHERI-mips.
-#if defined(__GNUC__) && !defined(__CHERI_PURE_CAPABILITY__) && !defined(CAPABILITY_VERSION)
+#if defined(__GNUC__)
+#ifdef __CHERI_PURE_CAPABILITY__
+	/*
+	 * Make sure the word-wise reads don't walk off the end
+	 * of caps with weakly-aligned ends.
+	 */
+	size_t space = __builtin_cheri_length_get(src) -
+	   __builtin_cheri_offset_get(src);
+	size_t offset = n - MIN(n, space);
```

Comment:
I'd call this `extra` or similar; `offset` makes it sound like it's at the start.